packages/core/src/agent/didcomm/versions/v2/sicpa-didcomm/SecretResolverService.ts

```diff
@@ -0,0 +1,55 @@
+import type { Secret, SecretsResolver } from 'didcomm'
+
+import { InjectionSymbols } from '../../../../../constants'
+import { KeyType } from '../../../../../crypto'
+import { inject, injectable } from '../../../../../plugins'
+import { Wallet } from '../../../../../wallet/Wallet'
+
+export enum KeyFormat {
+  Base58 = 'Base58',
+  Base64 = 'Base64',
+  JWK = 'JWK',
+  Multibase = 'Multibase',
+  Hex = 'Hex',
+  Pem = 'Pem',
+  BlockchainAccountId = 'BlockchainAccountId',
+  EthereumAddress = 'EthereumAddress',
+}
+
+const keyTypesMapping = {
+  [KeyType.Ed25519]: 'Ed25519VerificationKey2018',
+  [KeyType.X25519]: 'X25519KeyAgreementKey2019',
+  [KeyType.Bls12381g1g2]: 'Bls12381G1G2Key2020',
+  [KeyType.Bls12381g1]: 'Bls12381G1Key2020',
+  [KeyType.Bls12381g2]: 'Bls12381G2Key2020',
+}
```

Comment:
I think we already have a mapping of key types to verificadtio method types. Can we reuse it?